trieste/models/interfaces.py

```diff
@@ -320,14 +368,63 @@ def optimize(self, dataset: Dataset) -> None:
             model.optimize(Dataset(dataset.query_points, obs))
 
 
-class ReparametrizationSampler(ABC):
+class PredictJointModelStack(SupportsPredictJoint, ModelStack[SupportsPredictJoint]):
+    r"""
+    A :class:`PredictJointModelStack` is a wrapper around a number of
+    :class:`SupportsPredictJoint`\ s.
+    It delegates :meth:`predict_joint]` to each model.
+    """
+
+    def predict_joint(self, query_points: TensorType) -> tuple[TensorType, TensorType]:
+        r"""
+        :param query_points: The points at which to make predictions, of shape [..., B, D].
+        :return: The predictions from all the wrapped models, concatenated along the event axis in
+            the same order as they appear in :meth:`__init__`. If the wrapped models have predictive
+            distributions with event shapes [:math:`E_i`], the mean will have shape
+            [..., B, :math:`\sum_i E_i`], and the covariance shape
+            [..., :math:`\sum_i E_i`, B, B].
+        """
+        means, covs = zip(*[model.predict_joint(query_points) for model in self._models])
+        return tf.concat(means, axis=-1), tf.concat(covs, axis=-3)
+
+
+class TrainableSupportsPredictJoint(SupportsPredictJoint, TrainableProbabilisticModel):
```

Comment:
can we add a note for all of these empty classes for the PR in mypy on intersect development, once that is implemented we will be able to use intersect and all the classes directly where needed and remove these

Reply:
and perhaps raise an issue to remind us to keep an eye on this development?